apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/purchase-notice/index.jsx

```diff
@@ -1,19 +1,19 @@
 import { useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 import { store as noticesStore } from '@wordpress/notices';
-import { useEffect, useState } from 'react';
+import { useEffect, useRef } from 'react';
 import './style.scss';
 
 function PurchaseNotice() {
-	const [ hasPaymentNotice, setHasPaymentNotice ] = useState( false );
+	const hasPaymentNotice = useRef( false );
 	const { createNotice } = useDispatch( noticesStore );
 
 	useEffect( () => {
 		const noticePattern = /[&?]notice=([\w_-]+)/;
 		const match = noticePattern.exec( document.location.search );
 		const notice = match && match[ 1 ];
-		if ( 'purchase-success' === notice && hasPaymentNotice === false ) {
-			setHasPaymentNotice( true );
+		if ( 'purchase-success' === notice && hasPaymentNotice.current === false ) {
+			hasPaymentNotice.current = true;
 			createNotice(
 				'info',
 				__(
```

Comment:
Not directly related to this change, but we have a hardcoded "Pro plan" reference here that should probably also be fixed. While we could add extra logic to calculate what plan the user switched to specifically, it might be easier to just change this text to something like "Congrats! Premium blocks are now available to use."

Reply:
Updated!